src/main/java/com/desiato/puresynth/services/AuthenticationService.java

```diff
@@ -48,22 +48,23 @@ public Optional<Token> authenticate(AuthenticationRequestDTO request) {
         }
 
         String tokenValue = UUID.randomUUID().toString();
-        sessionRepository.save(new Session(tokenValue, user.getEmail(), user));
-        logger.info("Authentication succeeded and session created for email: {}", request.email());
+        Session newSession = new Session(tokenValue, user.getId());
+        sessionRepository.save(newSession);        logger.info("Authentication succeeded and session created for email: {}", request.email());
 
-        return Optional.of(new Token(tokenValue));
+        return Optional.of(new PureSynthToken(tokenValue));
     }
 
-    public boolean isUserAuthenticated(Token token) {
-        Optional<Session> sessionOpt = sessionRepository.findById(token.value());
+    public boolean isUserAuthenticated(PureSynthToken pureSynthToken) {
+        Optional<Session> sessionOpt = sessionRepository.findById(pureSynthToken.value());
 
         if (sessionOpt.isEmpty()) {
-            logger.warn("Token authentication failed: Session not found for token {}", token);
+            logger.warn("Token authentication failed: Session not found for token {}", pureSynthToken);
```

Comment:
Logging tokens is in general not recommended, as it creates a potential security threat.

Reply:
ok, thanks! yes makes sense